File: fastapi_server/app/schemas/stats.py
from pydantic import BaseModel, Field
from typing import List, Optional


class HabitStatsResponse(BaseModel):
    current_streak: int
    longest_streak: int
    total_success: int
    total_fail: int
    total_entries: int
    success_rate: int
    weekday: List[int]


class DayStateItem(BaseModel):
    date: str
    state: str
    value: Optional[float]


class RecentStatesResponse(BaseModel):
    habit_id: int
    days: List[DayStateItem]


class TodayHabitSummary(BaseModel):
    habit_id: int
    name: str
    emoji: str
    state: str
    current_streak: int


class TodaySummaryResponse(BaseModel):
    date: str
    scheduled: int
    success: int
    fail: int
    partial: int
    pending: int
    neutral: int
    habits: List[TodayHabitSummary]


class RecentRequest(BaseModel):
    habit_id: int
    days: int = Field(default=7, ge=1, le=365)

File: frontend_reactnative/src/api/stats.ts
/**
 * 统计 API
 */
import { request } from './client';
import type { HabitStats } from '../db/types';

export async function api_getHabitStats(habit_id: number): Promise<HabitStats> {
  return request<HabitStats>('/api/v1/stats/habit', { habit_id });
}

interface DayStateItem {
  date: string;
  state: string;
  value: number | null;
}

export async function api_getRecentStates(habit_id: number, days = 7): Promise<DayStateItem[]> {
  const res = await request<{ habit_id: number; days: DayStateItem[] }>('/api/v1/stats/recent', { habit_id, days });
  return res.days;
}

interface TodayHabitSummary {
  habit_id: number;
  name: string;
  emoji: string;
  state: string;
  current_streak: number;
}

export interface TodaySummary {
  date: string;
  scheduled: number;
  success: number;
  fail: number;
  partial: number;
  pending: number;
  neutral: number;
  habits: TodayHabitSummary[];
}

export async function api_getTodaySummary(): Promise<TodaySummary> {
  return request<TodaySummary>('/api/v1/stats/today-summary');
}
